pkg/auth/providers/genericoidc/genericoidc_provider.go

```diff
@@ -131,12 +131,18 @@ func (g *GenOIDCProvider) groupToPrincipal(groupName string) v3.Principal {
 func (g *GenOIDCProvider) getRedirectURL(config map[string]interface{}) string {
 	authURL, _ := baseoidc.FetchAuthURL(config)
 
-	return fmt.Sprintf(
+	redirectURL := fmt.Sprintf(
```

Comment:
Should we use [`url.JoinPath`](https://pkg.go.dev/net/url#JoinPath) here?

Reply:
@pmatseykanets no, because those are query params. It could be possible to use `url.Values` and the [`url.Values.Encode`](https://pkg.go.dev/net/url#Values.Encode).

```go
values := url.Values{}
values.Set("client_id", "myclient")
values.Set("response_type", "code")
values.Set("redirect_uri", "https://google.com")

redirectURL := fmt.Sprintf("%s?%s", authURL, values.Encode())
```